### ISSUE TITLE
[Toast] Add onPause / onResume props

### ISSUE THREAD

**Issue by nielsvdpol** [radix-ui/primitives#1453]
## Feature request
Add onPause / onResume props to Toast component so that when a progress bar is used, the animation can be paused.

_Ps. Couldn't find feature request jjenzz talked about in this issue: https://github.com/radix-ui/primitives/issues/1362#issuecomment-1126935808_

**Comment by la55u** [comment]
or something like `pauseOnHover` maybe?

**Comment by kylemh** [comment]
I saw in #1520 this was discusses as something unlikely to be done. I saw that #1477 doesn't quite resolve this. May I open a PR to sort it out?

**Comment by jjenzz** [comment]
@kylemh which part are you requesting to work on? benoit has asked in #1477 that the props be moved onto `Toast.Root` so when that feedback is addressed, you should be able to provide loading ui for your toasts.

**Comment by kylemh** [comment]
I'm hoping to address this issue (pause timers if a user is focused or has cursor hovering over a toast in the provider area).

Thinking about it more now that you've commented, I'm left wondering if global may actually be ideal. If a user is pausing one toast in a provider only showing one toast, global seems overkill; however, if we render multiple toasts, it wouldn't make sense to pause one toast and not the others.

Basically, I'd like to recreate Vercel's toast UI, but with some tweaks: https://vercel.com/design/toast

- Multiple toasts can render and create a stack
- There is a progress UI on the foreground toast to reflect the timer.
- If focus is on any toast or if the cursor is hovering any toast, the timer should be paused.

**Comment by jjenzz** [comment]
> If focus is on any toast or if the cursor is hovering any toast, the timer should be paused. (why this PR is needed)

the toast does already do this. please raise a reproducible GH issue separately if you are experiencing issues there tho.

this ticket is for exposing `onPause` and `onResume` props so that consumers can hook onto this behaviour when it happens in order to render timer UIs.

**Comment by kylemh** [comment]
amazing! 🎉 

> Pauses closing on hover, focus and window blur.

Should've seen that in the docs. Ignore me.
